chart/tests/test_elasticsearch_secret.py

```diff
@@ -132,3 +132,39 @@ def test_should_generate_secret_with_specified_schemes(self, scheme):
         )
 
         assert f"{scheme}://username:password@elastichostname:9200" == connection
+
+    def test_url_generated_when_user_is_empty(self):
+        connection = self._get_connection(
+            {
+                "elasticsearch": {
+                    "enabled": True,
+                    "connection": {"pass": "password", "host": "elastichostname", "port": 8080},
+                }
+            }
+        )
+
+        assert "http://elastichostname:8080" == connection
+
+    def test_url_generated_when_password_is_empty(self):
+        connection = self._get_connection(
+            {
+                "elasticsearch": {
+                    "enabled": True,
+                    "connection": {"user": "admin", "host": "elastichostname", "port": 8080},
+                }
+            }
+        )
+
+        assert "http://elastichostname:8080" == connection
+
+    def test_url_generated_with_valid_user_password(self):
+        connection = self._get_connection(
+            {
+                "elasticsearch": {
+                    "enabled": True,
+                    "connection": {"user": "admin", "pass": "pass", "host": "elastichostname", "port": 8080},
+                }
+            }
+        )
+
+        assert "http://admin:pass@elastichostname:8080" == connection
```

Comment:
You don't have a test where neither user/pass are set. It might be cleaner to parameterize these as well.

Reply:
Thanks, Im actually having problems passing JSON in parametrized.expand

``` @parameterized.expand(
        [
            # When user is empty

            ({"enabled": True, "connection" : {"pass": "password", "host": "elastichostname", "port": 8080}}),
            # When pass is empty
            # ({"connection": {"user": "admin", "host": "elastichostname", "port": 8080}}),
            # # When user and pass are empty
            # ({"connection": {"host": "elastichostname", "port": 8080}}),
        ],
    )```

Reply:
I was kinda thinking something like this (untested, but might help?):

```
@parameterized([
    ({}, ""),
    ({"user": "admin", ""),
    ({"pass": "password", ""),
    ({"user": "admin", "pass": "password", "admin:password"),
])
def test_something(self, extra_conn_kwargs, expected_userinfo):
    connection = self._get_connection({
           "elasticsearch": {
                "enabled": True,
                "connection": {
                        "host": "elastichostname",
                        "port": 8080,
                        **extra_conn_kwargs
                 }
            }
    })
    
    assert f"http://{expected_userinfo}@elastichostname:8080" == connection
````

I think not being a tuple might be your issue?

Reply:
Thanks , tuple definitely made the difference.